### PR TITLE
Add precommit for linter.gitlab-ci-jobs-codeowners

### PR DESCRIPTION
**Pull request by CelianR**
### What does this PR do?

> [!NOTE]
> Pre-push hooks for other new gitlab ci will come in a future PR

Adds a precommit hook for the gitlab linter `linter.gitlab-ci-jobs-codeowners`

### CODE REVIEW

**Review by KevinFairise2 — APPROVED** [review]
How much time does it take to run? If few seconds lgtm.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -116,6 +116,14 @@ repos:
...
       pass_filenames: false
       stages: [pre-push]
+    - id: gitlab-lint-jobs-codeowners
+      name: gitlab-lint-jobs-codeowners
+      description: lint the gitlab configuration to verify jobs codeowners
+      entry: 'inv linter.gitlab-ci-jobs-codeowners'
+      language: system
+      require_serial: true
+      files: .*gitlab.*\.yml$
+      pass_filenames: false
```
Comment (chouetz) [inline .pre-commit-config.yaml:126]: Should it be a `pre-push` (I don't know the duration of the linter)

Comment (CelianR) [reply .pre-commit-config.yaml:126]: This linter is quite fast so a pre-commit is ok (only modified files are checked). You can try `inv linter.gitlab-ci-jobs-codeowners --all-files` to lint all the files and it takes 0.25s

Comment (CelianR) [reply .pre-commit-config.yaml:126]: (but the other linters that are not added in this PR are much slower)

**Comment by CelianR** [comment]
> How much time does it take to run? If few seconds lgtm.

@KevinFairise2 0.25s for all the files

**Review by chouetz — APPROVED** [review]

**Pull request merged**